packages/slate-plugins/src/dnd/components/getSelectableElement.tsx

```diff
@@ -1,29 +1,49 @@
 import React, { forwardRef } from 'react';
 import { IStyleFunctionOrObject } from '@uifabric/utilities';
-import { RenderElementProps } from 'slate-react';
+import { Editor, Path } from 'slate';
+import { ReactEditor, RenderElementProps, useEditor } from 'slate-react';
 import { Selectable } from './Selectable';
 import {
   ElementWithId,
   SelectableStyleProps,
   SelectableStyles,
 } from './Selectable.types';
 
+export interface GetSelectabelElementOptions {
+  component: any;
+  styles?: IStyleFunctionOrObject<SelectableStyleProps, SelectableStyles>;
+  level?: number;
+  filter?: (editor: Editor, path: Path) => boolean;
+}
+
 export const getSelectableElement = ({
   component: Component,
   styles,
-}: {
-  component: any;
-  styles?: IStyleFunctionOrObject<SelectableStyleProps, SelectableStyles>;
-}) =>
-  forwardRef(({ attributes, element, ...props }: RenderElementProps, ref) => {
-    return (
-      <Selectable
-        attributes={attributes}
-        componentRef={ref}
-        element={element as ElementWithId}
-        styles={styles}
-      >
-        <Component attributes={attributes} element={element} {...props} />
-      </Selectable>
-    );
-  });
+  level,
+  filter,
+}: GetSelectabelElementOptions) => {
+  return forwardRef(
+    ({ attributes, element, ...props }: RenderElementProps, ref) => {
+      const editor = useEditor();
+      const path = ReactEditor.findPath(editor, element);
```

Comment:
Can you use `useMemo` here?

Reply:
Ah, good point. Will make these changes.